core/arch/arm/include/mm/core_mmu.h

```diff
@@ -86,6 +86,7 @@
  * MEM_AREA_IO_SEC:   Secure HW mapped registers
  * MEM_AREA_RES_VASPACE: Reserved virtual memory space
  * MEM_AREA_TA_VASPACE: TA va space, only used with phys_to_virt()
+ * MEM_AREA_TEE_RAM_UNCACHED:  Same as MEM_AREA_TEE_RAM, but cache disabled.
```

Comment:
Memory identifications go through `core_vbuf_is()/core_pbuf_is()`. there are also the mobj means for memory identification, mobj insteances being created based on known memory physical address ranges.

The fact is that there is a single "TEE RAM" defined in the core, and `core_pbuf/vuf_is(CORE_MEM_TEE_RAM, ...)` currently relies on this single area. I think adding a `TEE_RAM_UNCACHED` can be confusing. If your target "uncached TEE RAM" area targets some specific support, maybe its name should reflect this: TEE_RAM_LOWPOWER ?
Maybe you will be to split it into 3 pieces: read-only, read-write-noexec, read-only-exec. At Connect, there were discussions whether OP-TEE core should also split the TEE_RAM in such seperated map areas, as done when TEE pager is enable.